run-nextflow-tests/nfconfigtest

```diff
@@ -0,0 +1,207 @@
+                pipeline: Path,
+                testfile: Path) -> bool:
+    """
+    Run docker in a self-cleaning way and return the result.
+    """
+    pipeline = pipeline.resolve()
+    testfile = testfile.resolve()
+
+    output_file = testfile.with_name(testfile.stem + "-out.json")
+
+    container_id = subprocess.run(
+        [
+            "docker",
+            "create",
+            "--rm",
+            "-v", f"{pipeline}:{pipeline}",
+            image,
+            pipeline,
+            testfile.resolve()
+        ],
+        capture_output=True,
+        check=True
+    ).stdout.decode("utf-8").strip()
+
+    try:
+        subprocess.run(
+            ["docker", "start", "--attach", container_id],
+            check=True,
+            capture_output=True
+        )
+        # Remove the identical output file
+        output_file.unlink()
+        return True
+
+    except subprocess.CalledProcessError as err:
+        if err.returncode == 82:
+            logging.warning("  Failure")
+            logging.warning(
+                "  Compare these files and copy any expected changes:"
+            )
+            logging.warning("  Original: %s", testfile)
+            logging.warning("  Output:   %s", output_file)
+        else:
+            logging.error("Error running test!")
+            logging.error(err.stdout.decode('utf-8'))
+            logging.error(err.stderr.decode('utf-8'))
+            raise RuntimeError('Error running test!') from err
+    finally:
+        # Due to the the `docker create --rm` flag the container will delete
+        # itself once it stops, so make sure that it does so. This command will
+        # fail if the container has already stopped, so ignore the return code.
+        subprocess.run(
+            ["docker", "stop", container_id],
+            capture_output=True,
+            check=False
+        )
+
+    return False
+
+
+def run_tests(path: Path, dev_image: bool) -> int:
+    "Run the pipeline tests and returns the number of failures."
+    logging.info("Looking for tests in %s", path.resolve().name)
+
+    test_files = [
+        filepath for filepath in path.glob("**/configtest*.json")
+        if not filepath.stem.endswith("-out")
+        and ".mypy_cache" not in filepath.parts
+    ]
+
+    test_count = len(test_files)
+
+    if test_count == 0:
+        logging.info("No tests found!")
+        return 1
+
+    digits = math.ceil(math.log10(test_count))
+    success_count = 0
+
+    log_format = f"#%{digits}d/%d: %s"
+
+    for index, test in enumerate(sorted(test_files)):
+        logging.info(log_format, index+1, test_count, test.relative_to(path))
+
+        # Check the Nextflow version of the test
+        with test.open() as infile:
+            nextflow_version = json.load(infile)["nextflow_version"]
+
+        if nextflow_version != "23.10.0":
+            logging.error(
+                "  ERROR: Only Nextflow version 23.10.0 is supported"
+            )
+            continue
```

Comment:
Would it make sense to eventually update this to check all of the image tags available and then check that the Nextflow version requested is in the tags?

Reply:
Almost certainly! Right now 23.10.0 is hard-coded all over the place, but I want to handle that better in general.